Asthma/TabBarControllers/Dashboard/APHCalendarDataModel.m

```diff
@@ -38,45 +38,74 @@
 @implementation APHCalendarDataModel
 
 #pragma mark APHCalendarCollectionViewController delegate
--(NSDictionary *)userCompliedWithDailyScheduledTasks:(APHCalendarTaskType)task inMonth:(NSUInteger)month inYear:(NSUInteger)year{
-    
-    NSMutableDictionary *complianceDictionary = [[NSMutableDictionary alloc]init];
+-(void)createComplianceDictionaryForTaskType:(APHCalendarTaskType)task
+                                     inMonth:(NSUInteger)month
+                                      inYear:(NSUInteger)year
+{
     NSDateComponents *comps = [[NSDateComponents alloc] init];
     NSCalendar *gregorian = [[NSCalendar alloc]
                              initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
     APCDateRange *dateRange;
     
     if (task == kAPHCalendarTaskTypeParticipation) {
+        //should reflect the completion of the Daily Survey and Weekly Survey (when shown)
         
-        for (int day = 1; day < 32; day++) {
-            [comps setDay:day];
-            [comps setMonth:month];
-            [comps setYear:year];
-            
-            NSDate *date = [gregorian dateFromComponents:comps];
-            
-            if ([date earlierDate:[NSDate new]] == date) {
-                
-                dateRange = [[APCDateRange alloc]initWithStartDate:date durationInterval:86400];
-                
-                NSInteger scheduled = [APHCalendarDataModel allScheduledTasksForDateRange:dateRange completed:nil inContext:((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.mainContext].count;
-                
-                if (scheduled > 0) {
-                    
-                    NSInteger completedScheduled = [APHCalendarDataModel allScheduledTasksForDateRange:dateRange completed:[NSNumber numberWithBool:1] inContext:((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.mainContext].count;
-                    
-                    if ((float)completedScheduled / (float)scheduled > kParticipationTrophyThreshold){
-                        [complianceDictionary setObject:@"1" forKey:[NSString stringWithFormat:@"%i", day]];
-                    }else{
-                        [complianceDictionary setObject:@"0" forKey:[NSString stringWithFormat:@"%i", day]];
-                    }
-                }
-            }else{
-                return complianceDictionary;
-            }
+        [comps setDay:1];
+        [comps setMonth:month];
+        [comps setYear:year];
+        
+        NSDate *startDate = [gregorian dateFromComponents:comps];
+        
+        NSInteger endDay = 1;
+        
+        while ([self isValidDateForDay:endDay month:month year:year]) {
+            ++endDay;
         }
         
-        return complianceDictionary;
+        [comps setDay:endDay -1];
+        [comps setMonth:month];
+        [comps setYear:year];
+        NSDate *endDate = [gregorian dateFromComponents:comps];
```

Comment:
Some better ways to find the last day of a month (avoid a while loop):
http://stackoverflow.com/questions/2772348/getting-the-last-day-of-a-month